codepipeline_deploy.sh

```diff
@@ -12,6 +11,22 @@ if [ -z "${PIPELINE_TIMEOUT}" ]; then
     PIPELINE_TIMEOUT=1800
 fi
 
+# Get the HEAD commit ID for version.json in master branch if exists
+MASTER_COMMIT_ID=`aws codecommit get-file --repository-name AppManifest-"$SA_NAME"-"$ROS_DISTRO"-gazebo"$GAZEBO_VERSION" --file-path version.json`
+if [ $? -ne 0 ]; then
+    echo version.json does not yet exist in CC repository named "$CC_REPO_NAME"
+else
+    export MASTER_COMMIT_ID=`echo $MASTER_COMMIT_ID | jq -r '.commitId'`
+fi
+
+# Upload version.json to CodeCommit
+if [ -z "${MASTER_COMMIT_ID}" ]; then
```

Comment:
If `version.json` doesn't exist you're assuming the repository is empty. This breaks if we were to make changes to the repo creation process, for example if we start initializing the repo with a README file.

It'd be safer to `get-branch` instead of `get-file` (and change the json selector from `.commitId` to something like `.branch.commitId` but I think most of the logic can stay the same). That way the assumption would be: the repo is not empty iff the master branch exists. Not bulletproof but safer.

Reply:
I have remove use of codecommit to upload the version.json file. please take a look.

Reply:
See below for Tim's comments. We decided to use put-file and Tim cleaned up the code. Here is a successful build: https://travis-ci.org/aws-robotics/aws-robomaker-sample-application-helloworld/jobs/517940836